### PR TITLE
Non Secured APIs Feature

### PR DESCRIPTION
**Pull request by dinusha92**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1774

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/constants.go [hunk 1]
```diff
@@ -25,12 +25,15 @@ const (
...
-	xWso2Cors           string = "x-wso2-cors"
-	xThrottlingTier     string = "x-throttling-tier"
+	productionEndpoints  string = "x-wso2-production-endpoints"
+	sandboxEndpoints     string = "x-wso2-sandbox-endpoints"
+	xWso2BasePath        string = "x-wso2-basePath"
+	xWso2Label           string = "x-wso2-label"
+	xWso2Cors            string = "x-wso2-cors"
+	xThrottlingTier      string = "x-throttling-tier"
+	xAuthType            string = "x-auth-type"
+	xWso2DisableSecurity string = "x-wso2-disable-wso2"
```
Comment (Rajith90) [inline adapter/internal/oasparser/model/constants.go:35]: this extension seems wrong. Could we please check the mgw 3.2.0 docs and get the extension from there

File: adapter/internal/oasparser/config_generator.go [hunk 2]
```diff
@@ -127,17 +127,18 @@ func GetEnforcerAPI(mgwSwagger model.MgwSwagger, lifeCycleState string) *api.Api
...
+		Description:       mgwSwagger.GetDescription(),
+		BasePath:          mgwSwagger.GetXWso2Basepath(),
+		Version:           mgwSwagger.GetVersion(),
+		ProductionUrls:    prodUrls,
+		SandboxUrls:       sandUrls,
+		Resources:         resources,
+		ApiLifeCycleState: lifeCycleState,
+		Tier:              mgwSwagger.GetXThrottlingTier(),
+		SecurityScheme:    mgwSwagger.GetSetSecurityScheme(),
+		AuthType:          mgwSwagger.GetXAuthType(),
```
Comment (Rajith90) [inline adapter/internal/oasparser/config_generator.go:141]: We need to have this in operation level , not only in the API level

File: enforcer/src/main/java/org/wso2/micro/gateway/enforcer/util/FilterUtils.java [hunk 3]
```diff
@@ -173,6 +173,32 @@ public static String getTenantDomainFromRequestURL(String requestURI) {
...
+        authContext.setUsername(APIConstants.END_USER_ANONYMOUS);
+        //authContext.setApiTier(APIConstants.UNAUTHENTICATED_TIER);
+        authContext.setApplicationId(clientIP);
+        authContext.setApplicationName(null);
+        authContext.setApplicationTier(APIConstants.UNAUTHENTICATED_TIER);
+        authContext.setSubscriber(clientIP);
+        authContext.setApiName(requestContext.getMatchedAPI().getAPIConfig().getName());
+        authContext.setStopOnQuotaReach(true);
+        authContext.setConsumerKey(null);
+        authContext.setCallerToken(null);
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/micro/gateway/enforcer/util/FilterUtils.java:197]: In synapse gateway do we use null values here

Comment (dinusha92) [reply enforcer/src/main/java/org/wso2/micro/gateway/enforcer/util/FilterUtils.java:197]: Yes. Current implementation (https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/oauth/OAuthAuthenticator.java#L257-L305)

File: adapter/internal/oasparser/config_generator.go (hunk 2 above)
Comment (dinusha92) [reply adapter/internal/oasparser/config_generator.go:141]: We have thin in operational level as well. Please refer GetEnforcerAPIOperation method.

File: adapter/internal/oasparser/model/constants.go (hunk 1 above)
Comment (dinusha92) [reply adapter/internal/oasparser/model/constants.go]: Fixes type https://github.com/wso2/product-microgateway/pull/1775/commits/0da4349b091d28e12161105106bf08fb3c279637

File: api/wso2/discovery/api/Resource.proto [hunk 4]
```diff
@@ -31,6 +31,7 @@ message Operation {
     string method = 1;
     repeated SecurityList security = 2;
     string tier = 3;
+    string authType = 4;
```
Comment (dinusha92) [inline api/wso2/discovery/api/Resource.proto:34]: refactor to enableAuth

**Comment by dinusha92** [comment]
Closing the PR as a new PR will be sent.

**Pull request closed without merging**